SmartDeviceLink/SDLSoftButtonObject.m

```diff
@@ -60,6 +60,23 @@ - (BOOL)transitionToState:(NSString *)stateName {
     return YES;
 }
 
+- (void)transitionToNextState {
+    NSString *nextStateName = nil;
+    for (NSUInteger i = 0; i < self.states.count; i++) {
+        if ([self.states[i].name isEqualToString:self.currentStateName]) {
+            NSUInteger nextStateNumber = (i == self.states.count) ? 0 : i;
+            nextStateName = self.states[nextStateNumber].name;
+            break;
+        }
+    }
```

Comment:
State transitions don't occur because the index is not incremented.

Reply:
😅 fixed